### PR TITLE
tests: align pytest version across requirements

### PR DESCRIPTION
**Pull request by guybortnikov**
## Description
Align `pytest` across pip requirements files

### CODE REVIEW

**Comment by guybortnikov** [comment]
> Is this file even needed? If there was a mismatch but CI & build pass, is it even used? One of the 2 is not required I presume


I was afraid.. now removed from `dev-requirements`, let's see CI passes

**Comment by Jongy** [comment]
I don't think it'll pass because the tests don't install dev-requirements.txt. For all I care you can merge both into dev-requirements.txt and install it in `test.sh`.

File: test-requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-pytest==6.2.4
 PyYAML==6.0
 docker==5.0.0
 six==1.16.0
```
Comment (Jongy) [inline test-requirements.txt:3]: Let's just remove this file and move the other things to `dev-requirements.txt`. It's not needed & confusing

Comment (guybortnikov) [reply test-requirements.txt]: done

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,8 +1,11 @@
-pytest==7.0.1
+pytest==7.0.1 # pytest types are needed for linting (dev process)
```
Comment (Jongy) [inline dev-requirements.txt:1]: Comment irrelevant anymore?

**Review by Jongy — APPROVED** [review]
Can merge once tests pass

**Pull request merged**